infinitest-runner/src/main/java/org/infinitest/config/InfinitestConfigurationParser.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Infinitest, a Continuous Test Runner.
+ *
+ * Copyright (C) 2010-2013
+ * "Ben Rady" <benrady@gmail.com>,
+ * "Rod Coffin" <rfciii@gmail.com>,
+ * "Ryan Breidenbach" <ryan.breidenbach@gmail.com>
+ * "David Gageot" <david@gageot.net>, et al.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.infinitest.config;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.CharSource;
+
+/**
+ * Parses the content of the infinitest.filters file
+ * @author sarod
+ *
+ */
+public class InfinitestConfigurationParser {
+
+	private static final String LEGACY_SUFFIX = "\\s?=\\s?(.+)";
+	private static final String LEGACY_PREFIX = "^\\s*#+\\s?";
+
+	private static final Splitter ON_COMMA_SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+
+	private interface LineParser {
+
+		/**
+		 * Process the line if relevant
+		 * 
+		 * @param line
+		 * @return true if the line was sucssefully parsed and processed by this
+		 *         parser
+		 */
+		boolean maybeProcessLine(String line);
```

Comment:
Optional: I usually used processLineIfRelevant naming more than maybeProcessLine. it seems to be more a matter of taste. maybeProcessLine seems fine too.